aperturedb/ParallelQuery.py

```diff
@@ -138,10 +138,34 @@ def __init__(self, db, dry_run=False):
 
     def generate_batch(self, data):
         """
-            Here we flatten the individual queries to run them as
-            a single query in a batch
+        Here we flatten the individual queries to run them as
+        a single query in a batch
+        We also update the _ref values and connections refs.
         """
-        q = [cmd for query in data for cmd in query[0]]
+        def update_refs(batched_commands):
+            updates = {}
+            for i, cmd in enumerate(batched_commands):
+                if isinstance(cmd, list):
+                    # Only pralllel queries will work.
+                    break
+                values = cmd[list(cmd.keys())[0]]
+                if "_ref" in values:
+                    updates[values["_ref"]] = i + 1
+                    values["_ref"] = i + 1
+                    assert values["_ref"] < 100000
+                if "image_ref" in values:
+                    values["image_ref"] = updates[values["image_ref"]]
+                if "video_ref" in values:
+                    values["video_ref"] = updates[values["video_ref"]]
+                if "is_connected_to" in values and "_ref" in values["is_connected_to"]:
+                    values["is_connected_to"]["_ref"] = updates[values["is_connected_to"]["_ref"]]
+                if "src" in values:
+                    values["src"] = updates[values["src"]]
+                if "dst" in values:
+                    values["dst"] = updates[values["dst"]]
+            return batched_commands
+
+        q = update_refs([cmd for query in data for cmd in query[0]])
```

Comment:
This looks good - is there any reason to keep the "video" and "image" since PQ is just doing matching at this point?

Reply:
not sure if I understand this correctly, but this would serve the case where getitem generates a transaction with both video and image, and entities with some connected to video, and others to images.

Reply:
right - should we just change "video" and 'image" into something more generic?